sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java

```diff
@@ -0,0 +1,170 @@
+        }
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (mDevice != null) {
+            try {
+                // power down device
+                mDevice.writeRegByte(LPS_CTRL_REG1, (byte) 0);
+            } catch (Exception any) {
+                // we tried
+            }
+            try {
+                mDevice.close();
+            } finally {
+                mDevice = null;
+            }
+        }
+    }
+
+    private int readSigned24(int a0, int a1, int a2) throws IOException {
+        int ret = (mDevice.readRegByte(a0) & 0xFF);
+        ret += ((int) mDevice.readRegByte(a1) & 0xFF) << 8;
+        ret += ((int) mDevice.readRegByte(a2) & 0xFF) << 16;
+        if (ret < 8388608) return ret;
+        else return ret - 16777216;
+    }
+
+    private int readSigned16(int a0, int a1) throws IOException {
+        int ret = (mDevice.readRegByte(a0) & 0xFF);
+        ret += (mDevice.readRegByte(a1) & 0xFF) << 8;
+        if (ret < 32768) return ret;
+        else return ret - 65536;
+    }
+
+    /**
+     * The sensor seems to have an offset to the actual pressure. You can find your local "real" pressure quite easily on the web. Get the measured value from the
+     * sensor and compute the difference. The value obtained can be passed to this method to "calibrate" your board's sensor.
+     *
+     * @param hPa difference to actual air pressure
+     * @throws IOException from I2cDevice
+     */
+    public void setBarometerOffset(double hPa) throws IOException {
+        this.milliBarAdjust = (int) Math.round(hPa * 4096);
+    }
+
+    /**
+     * Fetch raw value, see the data sheet. Note that this call waits for data to be available.
+     *
+     * @return The raw sensor value, adjusted by the given offset (if any).
+     * @throws IOException from I2cDevice
+     */
+    public int getBarometerRaw() throws IOException {
+        // wait for data available
+        while (0 == (mDevice.readRegByte(LPS_STATUS_REG) & 2)) {
+            try {
+                Thread.sleep(1);
+            } catch (InterruptedException e) {
+                throw new IOException(e);
+            }
+        }
+        return readSigned24(LPS_PRESS_OUT_XL, LPS_PRESS_OUT_L, LPS_PRESS_OUT_H) + milliBarAdjust;
+    }
+
+    /**
+     * Fetch raw value, see the data sheet. Note that this call waits for data to be available.
+     *
+     * @return The raw sensor value.
+     * @throws IOException from I2cDevice
+     */
+    public int getTemperatureRaw() throws IOException {
+        // wait for data available
+        while (0 == (mDevice.readRegByte(LPS_STATUS_REG) & 1)) {
+            try {
+                Thread.sleep(1);
+            } catch (InterruptedException e) {
+                throw new IOException(e);
+            }
+        }
+        return readSigned16(LPS_TEMP_OUT_L, LPS_TEMP_OUT_H);
+    }
+
+    /**
+     * Fetch air pressure in hPa (milli bar). Note that this call waits for data to be available.
```

Comment:
I believe it's just millibar, without a space.

Reply:
Fixed.